arangod/Aql/Query.h

```diff
@@ -206,6 +206,13 @@ class Query : public QueryContext {
   aql::SnippetList& snippets() { return _snippets; }
   aql::ServerQueryIdList& serverQueryIds() { return _serverQueryIds; }
   aql::ExecutionStats& executionStats() { return _execStats; }
+
+
+  // Debug method to kill a query at a specific position
+  // during execution. It internally asserts that the query
+  // is actually visible through other APIS (e.g. current queries)
+  // so user actually has a chance to kill it here.
+  void debugKillQuery() override;
```

Comment:
Wondering if we should wrap this method and its implementation into a 
```
#ifdef ARANGODB_ENABLE_FAILURE_TESTS
...
#endif
```
as well.

Reply:
I would love that (and initial did so), BUT:
You cannot compile without FAILURE tests anymore.

Here is why:

```
TRI_IF_FAILURE("foo") {
  query->debugKillQuery();
}
```

will be translated to:
```
if (false) {
  query->debugKillQuery()
}
```
without FAILURE_TESTS

And compilers are not clever enough to figure out that this code is in-fact unreachable and will complain that the function does not exist.

I tried

```
if constexpr (false) {
```

as well here, but this only works within templated code, otherwise it will also try to compile the unreachable branch.

So we need some way to reformulate the TRI_IF_FAILURE macro to remove the non-compiled code first.

Reply:
Or alternatively wrap all users of `TRI_IF_FAILURE(...) { ... }` into `#ifdef`s as well.
This is also not a great alternative.

Ok, then let's leave it as is.